pkg/apis/cassandra.rook.io/v1alpha1/types.go

```diff
@@ -0,0 +1,91 @@
+package v1alpha1
+
+import (
+	rook "github.com/rook/rook/pkg/apis/rook.io/v1alpha2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ***************************************************************************
+// IMPORTANT FOR CODE GENERATION
+// If the types in this file are updated, you will need to run
+// `make codegen` to generate the new types under the client/clientset folder.
+// ***************************************************************************
+
+// Kubernetes API Conventions:
+// https://github.com/kubernetes/community/blob/af5c40530f50c3b36c13438187b311102093ede5/contributors/devel/api-conventions.md
+// Applicable Here:
+//  * Optional fields use a pointer to correctly handle empty values.
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type Cluster struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              ClusterSpec   `json:"spec"`
+	Status            ClusterStatus `json:"status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type ClusterList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []Cluster `json:"items"`
+}
+
+// ClusterSpec is the desired state for a Cassandra Cluster.
+type ClusterSpec struct {
+	// Version of Cassandra to use.
+	Version string `json:"version"`
+	// Repository to pull the image from.
+	Repository *string `json:"repository,omitempty"`
+	// Datacenter that will make up this cluster.
+	Datacenter DatacenterSpec `json:"datacenter"`
+	// User-provided image for the sidecar that replaces default.
+	SidecarImage *ImageSpec `json:"sidecarImage,omitempty"`
+}
+
+// ClusterStatus is the status of a Cassandra Cluster
+type ClusterStatus struct {
+	Racks map[string]RackStatus
+}
+
+// RackStatus is the status of a Cassandra Rack
+type RackStatus struct {
+	// ReadyInstances is the number of ready instances in the specific Rack
+	ReadyMembers int32
```

Comment:
would total number of members be a useful field here too? or maybe that's retrievable from the `RackSpec`

Reply:
It would ! 
I added a new field called members.